### PR TITLE
bugfix/21698-tooltip outside-exceeds-viewport

### LINKED ISSUE #21698: when tooltip.outside is true then if hover then we are getting the scroll and tooltip is not visible fully
#### Expected behaviour
  we should not get scroll and see the full tooltip

#### Actual behaviour
getting scroll

#### Live demo with steps to reproduce

https://jsfiddle.net/kluvin/xthgz2jk/9

#### Product version

"highcharts": "^11.3.0"

#### Affected browser(s)
all

### PR DESCRIPTION
**Pull request by MarkusBarstad**
Fixed #21698, tooltip with outside could exceed past the viewport.

Issue seems to be that when `text` is given to `tooltip label`, the resulting `bBox` is too large.

### CODE REVIEW

**Review by bre1470 — APPROVED** [review]
Fantastic! 👍

File: ts/Core/Tooltip.ts [hunk 1]
```diff
@@ -1094,8 +1094,8 @@ class Tooltip {
                             x: clamp(
                                 label.x || 0,
                                 0,
-                                this.getPlayingField().width -
-                                    (label.width || 0)
+                                (this.getPlayingField().width) -
```
Comment (TorsteinHonsi) [inline ts/Core/Tooltip.ts:1097]: Unnecessary parentheses

File: ts/Core/Tooltip.ts [hunk 2]
```diff
@@ -1798,10 +1798,23 @@ class Tooltip {
...
+
+            const baseWidth = width + pad;
 
             renderer.setSize(
-                width + pad,
+                this.outside ? clamp(
+                    baseWidth,
+                    0,
+                    // 'Outside' container always has 'left: 1px;'
+                    H.doc.documentElement.clientWidth - 1
```
Comment (TorsteinHonsi) [inline ts/Core/Tooltip.ts:1815]: Use the local `doc`
```suggestion
                    doc.documentElement.clientWidth - 1
```

**Review by TorsteinHonsi — CHANGES_REQUESTED** [review]
Thanks, nice solution! Just a little nitpicking.

File: ts/Core/Tooltip.ts [hunk 3]
```diff
@@ -1798,10 +1799,23 @@ class Tooltip {
...
+                2 *
+                distance +
+                2
+            );
+
+            const baseWidth = width + pad;
 
             renderer.setSize(
-                width + pad,
+                this.outside ? clamp(
```
Comment (TorsteinHonsi) [inline ts/Core/Tooltip.ts:1812]: Wait a minute, found another nit to pick 😬 . The `renderer` [only exists](https://github.com/highcharts/highcharts/blob/a09cb5c5fe2ebe820e6f32be953cb9a42c6a31fd/ts/Core/Tooltip.ts#L475-L483) when `this.outside` is true. Which means we can skip the ternary here and always use the clamp.

**Review by TorsteinHonsi — CHANGES_REQUESTED** [review]

File: ts/Core/Tooltip.ts (hunk 3 above)
Comment (MarkusBarstad) [reply ts/Core/Tooltip.ts:1812]: Good catch!

**Review by TorsteinHonsi — APPROVED** [review]

**Pull request merged**